### PR TITLE
Use JSON.stringify.

### LINKED ISSUE #69: Show string implementation
When showing a string it just returns a raw string. However, when you want to use the show implementation in a larger data structure, it doesn't really work too well. For instance

```javascript
var Ident = function(a) {
  this.val = a;
}
var IdentShow = function(a) {
  return function(id) {
    return "Ident(" + a.show(id.val) + ")";
  };
}
```

If you make an `Ident string` when it goes to show it, sometimes it will show `Ident()`, if the generated string was the empty string for instance. What seems more natural would be to see `Ident("")` or something indicating that the string was empty.

It seems like it is enough to just change these lines:

https://github.com/jsverify/jsverify/blob/67a97152fc91b1d4ace6b2d8463e06d132dad717/lib/primitive.js#L252
https://github.com/jsverify/jsverify/blob/67a97152fc91b1d4ace6b2d8463e06d132dad717/lib/primitive.js#L263
https://github.com/jsverify/jsverify/blob/67a97152fc91b1d4ace6b2d8463e06d132dad717/lib/primitive.js#L273
https://github.com/jsverify/jsverify/blob/67a97152fc91b1d4ace6b2d8463e06d132dad717/lib/primitive.js#L287
https://github.com/jsverify/jsverify/blob/67a97152fc91b1d4ace6b2d8463e06d132dad717/lib/primitive.js#L296

to be like this one:

https://github.com/jsverify/jsverify/blob/67a97152fc91b1d4ace6b2d8463e06d132dad717/lib/primitive.js#L309-L311

Thoughts?

### PR DESCRIPTION
**Pull request by joneshf**
Fixes #69

### CODE REVIEW

File: lib/show.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ var utils = require("./utils.js");
   - `show.def(x : a): string`
 */
 function showDef(obj) {
-  return "" + obj;
+  return JSON.stringify(obj);
```
Comment (StoneCypher) [inline lib/show.js]: I don't actually know what you're using this for, so I should probably keep my mouth shut.

But is it acceptable that 1) this will drop some stuff, like functions, from the output, and 

2) that this will break in chrome if obj has circular references?

Comment (joneshf) [inline lib/show.js]: I appreciate the review. I'm not sure either of those is acceptable. Maybe I should just leave that one definition alone and only stringify the strings? Does that kill anything else?

Comment (StoneCypher) [inline lib/show.js]: I don't have a good answer for you.  I'm only just barely getting started in `jsverify`.

Comment (phadej) [inline lib/show.js]: Circularity and functions aren't the problem. For those you have to define your own `show` anyway, shouldn't rely on default (have to check function case though).

File: Makefile [hunk 2]
```diff
@@ -5,7 +5,7 @@ BINDIR=node_modules/.bin
 JSHINT=$(BINDIR)/jshint
 ESLINT=$(BINDIR)/eslint
 JSCS=$(BINDIR)/jscs
-MOCHA=$(BINDIR)/_mocha
+MOCHA=$(BINDIR)/mocha
```
Comment (phadej) [inline Makefile]: why this change. Is istanbul still working?

File: .eslintrc [hunk 3]
```diff
@@ -3,6 +3,6 @@
     "node": true
   },
   "rules": {
-    "no-comma-dangle": 0
+    "comma-dangle": 0
```
Comment (phadej) [inline .eslintrc]: The bump of eslint version should be in-place too?

File: Makefile (hunk 2 above)
Comment (joneshf) [inline Makefile]: Something happened with `mocha` at `2.2.0` and `_mocha` doesn't work anymore. Everything else seems to work with `mocha`, but I encourage you to check before merging.

File: lib/show.js (hunk 1 above)
Comment (joneshf) [inline lib/show.js]: Oh, so then leave this?

**Comment by phadej** [comment]
Merged in https://github.com/jsverify/jsverify/pull/73. Thanks @joneshf for PR
